### ISSUE TITLE
[BUG] Rule not triggered for METHOD_CALL with specific object and parameter type

### ISSUE THREAD

**Issue by ibragins** [konveyor/analyzer-lsp#853]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Konveyor version

latest

### Priority

Undefined (Default)

### Current Behavior

```
  when:
    java.referenced:
      location: METHOD_CALL
      pattern: org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean.setJpaVendorAdapter(org.springframework.orm.jpa.vendor.HibernateJpaVendorAdapter)
```
Should trigger an incident in the following code (extracted from [tackle-testapp](https://github.com/konveyor/tackle-testapp-public/blob/main/src/main/java/io/konveyor/demo/ordermanagement/config/PersistenceConfig.java#L35))

`em.setJpaVendorAdapter(new HibernateJpaVendorAdapter());`

But it doesn't trigger any incidents.

### How Reproducible

Always (Default)

### Environment

```markdown
- **Analyzer Image SHA** 
- **JDTLS  base image SHA**
- **Provider Settings JSON**
```
